LibOS/shim/src/bookkeep/shim_vma.c

```diff
@@ -37,229 +37,327 @@
 #include <errno.h>
 #include <stdbool.h>
 
-unsigned long mem_max_npages __attribute_migratable = DEFAULT_MEM_MAX_NPAGES;
-
-static void * heap_top, * heap_bottom;
-
-#define VMA_MGR_ALLOC   64
+/*
+ * Internal bookkeeping for VMAs (virtual memory areas). This data
+ * structure can only be accessed in this source file, with vma_list_lock
+ * held. No reference counting needed in this data structure.
+ */
+DEFINE_LIST(shim_vma);
+struct shim_vma {
```

Comment:
My understanding is vma contains [start, end). if that's true it's worth the a comment here.

Reply:
+1

Reply:
LGTM now.